src/core/data/image/image_type.cpp

```diff
@@ -75,4 +75,18 @@ const unsigned char* image_type::get_image_data() const {
   }
 }
 
+// The equality behavior is currently mirrored from python implementation 
+// where every property is compared directly, but we might want to later
+// support comparing two images with different lossless types.
+bool image_type::operator==(const image_type& other) const {
+  return
+      m_height == other.m_height &&
+      m_width == other.m_width &&
+      m_channels == other.m_channels && 
+      m_image_data_size == other.m_image_data_size &&
+      m_version == other.m_version &&
+      m_format == other.m_format &&
+      strcmp(m_image_data.get(), other.m_image_data.get()) == 0;
```

Comment:
This works only with C style strings, which means it will stop at the first zero byte.  Use std::equal instead:
`std::equal(m_image_data.get(), m_image_data.git() + m_image_data_size, other.m_image_data.get())`

Reply:
Thanks for pointing out! I was not aware of the EOF character that could potentially be in the char array.